### PR TITLE
Update migration snapshot task

### PR DESCRIPTION
**Pull request by dakota**
Updates the migration snapshot task to take into account non-default primary key fields.

### CODE REVIEW

File: src/View/Helper/MigrationHelper.php [hunk 1]
```diff
@@ -113,6 +113,22 @@ public function columns($table)
...
+     * @return array
+     */
+    public function primaryKey($table) {
+        $collection = $this->config('collection');
+        $tableSchema = $collection->describe($table);
+        foreach ($tableSchema->columns() as $column) {
+            if ($tableSchema->primaryKey() == [$column]) {
+                return ['name' => $column, 'info' => $this->column($tableSchema, $column)];
+            }
+        }
```
Comment (markstory) [inline src/View/Helper/MigrationHelper.php]: What about composite keys?

File: src/View/Helper/MigrationHelper.php [hunk 2]
```diff
@@ -113,6 +113,22 @@ public function columns($table)
...
+     * Returns the primary key data for a given table
+     *
+     * @param string $table Name of the table ot retrieve primary key for
+     * @return array
+     */
+    public function primaryKey($table) {
+        $collection = $this->config('collection');
+        $tableSchema = $collection->describe($table);
+        foreach ($tableSchema->columns() as $column) {
+            if ($tableSchema->primaryKey() == [$column]) {
```
Comment (dereuromark) [inline src/View/Helper/MigrationHelper.php]: Can be be strict here, as well?

File: src/Shell/Task/MigrationSnapshotTask.php [hunk 3]
```diff
@@ -79,7 +79,7 @@ public function templateData()
             'collection' => $collection,
             'tables' => $tables,
             'action' => 'create_table',
-            'name' => $this->BakeTemplate->viewVars['name'],
+            'name' => $this->Template->viewVars['name'],
```
Comment (josegonzalez) [inline src/Shell/Task/MigrationSnapshotTask.php]: Seems like this broke things.

File: src/View/Helper/MigrationHelper.php (hunk 1 above)
Comment (dakota) [inline src/View/Helper/MigrationHelper.php]: Forgot about those, will need to take them into account.

File: src/Shell/Task/MigrationSnapshotTask.php (hunk 3 above)
Comment (dakota) [inline src/Shell/Task/MigrationSnapshotTask.php]: Looks like tests need to be updated too.

Comment (josegonzalez) [inline src/Shell/Task/MigrationSnapshotTask.php]: It worked the old way - which is how the bake plugin handled it.

Comment (dakota) [inline src/Shell/Task/MigrationSnapshotTask.php]: The old way `BakeTemplate` was giving me undefined object errors and leaving me with a snapshot with no class name. Bake plugin seems to have changed to use `Template`.

Comment (AD7six) [inline src/Shell/Task/MigrationSnapshotTask.php]: @dakota I suspect your version of bake is out of date.

[BakeTemplate](https://github.com/cakephp/bake/blob/master/src/Shell/Task/SimpleBakeTask.php#L31) is the task for rendering bake templates, [Template](https://github.com/cakephp/bake/blob/master/src/Shell/Task/TemplateTask.php) is the task for creating application/plugin template files (it used to be called View).

Comment (dakota) [inline src/Shell/Task/MigrationSnapshotTask.php]: @AD7six You suspect right! Didn't notice that bake was on `0.1.x`. Will update!

File: src/View/Helper/MigrationHelper.php (hunk 1 above)
Comment (dakota) [inline src/View/Helper/MigrationHelper.php]: @markstory Added support for composite keys.

**Pull request merged**